scripts/prism_downloader.py

```diff
@@ -24,78 +36,96 @@
 DEFAULT_PARAMS = ["tmean", "tmax", "tmin", "ppt", "vpdmax", "vpdmin", "tdmean"]
 
 
-def setupArgs() -> None:
+def configure_spatial_env(proj_dir: str | None, gdal_dir: str | None) -> None:
+  if proj_dir:
+    os.environ["PROJ_LIB"] = proj_dir
+    try:
+      pyproj_datadir.set_data_dir(proj_dir)
+    except Exception:
+      pass
+  if gdal_dir:
+    os.environ["GDAL_DATA"] = gdal_dir
+  if proj_dir or gdal_dir:
+    os.environ.setdefault("PROJ_NETWORK", "ON")
+
+
+def setupArgs() -> argparse.Namespace:
   parser = argparse.ArgumentParser(
-    description="""Download PRISM downsampled data and clip to region and save as zarr. See https://www.prism.oregonstate.edu/ and https://www.prism.oregonstate.edu/documents/PRISM_datasets.pdf
-                                     Uses the prism Webservice -- https://prism.oregonstate.edu/documents/PRISM_downloads_web_service.pdf"""
+    description=(
+      "Download PRISM downsampled data, clip to region, and save as Zarr. "
+      "See https://www.prism.oregonstate.edu/ for dataset details."
+    )
   )
   parser.add_argument(
     "--parameters",
     type=str,
     default=DEFAULT_PARAMS,
     choices=DEFAULT_PARAMS,
     nargs="+",
-    help="the variables that will be downloaded - these are PRISM output vars and defaults to all",
-  )
-  parser.add_argument(
-    "--startDate",
-    type=str,
-    required=True,
-    help="Start date of data to download e.g. 2020-10-01 for October 1, 2020",
-  )
-  parser.add_argument(
-    "--endDate",
-    type=str,
-    required=True,
-    help="End date of data to download e.g. 2020-10-01 for October 1, 2020",
+    help="PRISM variables to download (default: all).",
   )
+  parser.add_argument("--startDate", type=str, required=True, help="Start date e.g. 2020-10-01")
+  parser.add_argument("--endDate", type=str, required=True, help="End date e.g. 2020-10-02")
   parser.add_argument(
     "--outputDir",
     default="data/weather_data/",
     type=str,
-    help="Directory/path to download data/output zarr to.",
+    help="Directory/path to download data/output Zarr to.",
   )
   parser.add_argument(
     "--geojson",
     default="data/GIS/SkagitBoundary.json",
     type=str,
-    help="Path to/name of geo_json file that geogrpahically limits the downloaded data",
+    help="GeoJSON file limiting the downloaded data.",
   )
   parser.add_argument(
     "--region",
     default=DEFAULT_REGION,
     type=str,
     choices=REGION_OPTIONS,
-    help="Region to download data for. Default is US",
+    help="Region to download data for.",
   )
   parser.add_argument(
     "--resolution",
     default=DEFAULT_RESOLUTION,
     type=str,
     choices=RESOLUTION_OPTIONS,
-    help="Resolution of data to download. Default is 4km. Only 4km and 800m are available.",
+    help="Data resolution.",
```

Comment:
```suggestion
    help="Resolution of data to download. Default is 4km. Only 4km and 800m are available.,
```

The original seems better here

Reply:
My bad, sorry, I didnt notice the description getting changed